src/lexer/Lexer.ts

```diff
@@ -680,21 +672,18 @@ export class Lexer {
                 this.advance();
             }
 
-            asString = this.source.slice(this.start, this.current);
             this.addToken(TokenKind.FloatLiteral);
-            return;
         } else if (containsDecimal) {
             // anything with a decimal but without matching Double rules is a Float
             this.addToken(TokenKind.FloatLiteral);
-            return;
-        }
-
-        if (designator === '&') {
+        } else if (designator === '&') {
             // numeric literals ending with "&" are forced to LongIntegers
-            asString = this.source.slice(this.start, this.current);
             this.advance();
             this.addToken(TokenKind.LongIntegerLiteral);
-
+        } else if (designator === '%') {
```

Comment:
This is the core of the change. we just needed a separate if block for handling `%`